src/wrapper/java_vm/init_args.rs

```diff
@@ -0,0 +1,126 @@
+use std::os::raw::c_void;
+
+use std::ffi::CString;
+
+use sys::{
+    JavaVMInitArgs,
+    JavaVMOption,
+};
+
+use JNIVersion;
+
+error_chain! {
+    errors {
+        /// Opt string had internal null
+        NullOptString(opt: String) {
+            display("internal null in option: {}", opt)
+            description("internal null in option string")
+        }
+    }
+}
+
+/// Builder for JavaVM InitArgs
+pub struct InitArgsBuilder {
+    opts: Vec<String>,
+    ignore_unrecognized: bool,
+    version: i32,
+}
+
+impl Default for InitArgsBuilder {
+    fn default() -> Self {
+        InitArgsBuilder {
+            opts: vec![],
+            ignore_unrecognized: false,
+            version: JNIVersion::V1.into(),
+        }
+    }
+}
+
+impl InitArgsBuilder {
+    /// Create a new default InitArgsBuilder
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Add an option to the init args
+    ///
+    /// The `vfprintf`, `abort`, and `exit` options are unsupported at this
+    /// time.
+    pub fn option(self, opt_string: &str) -> Self {
+        let mut s = self;
+
+        match opt_string {
+            "vfprintf" | "abort" | "exit" => return s,
+            _ => {}
+        }
+
+        s.opts.push(opt_string.into());
+
+        s
+    }
+
+    /// Set JNI version for the init args
+    ///
+    /// Default: V1
+    pub fn version(self, version: JNIVersion) -> Self {
+        let mut s = self;
+        s.version = version.into();
+        s
+    }
+
+    /// Set the `ignoreUnrecognized` init arg flag
+    ///
```

Comment:
May add a clarification from the docs:

«If `ignoreUnrecognized` is true, `JavaVM::new` ignores all unrecognized option strings that begin with "-X" or "_". If `ignoreUnrecognized` is false, `JavaVM::new` returns `Err` as soon as it encounters any unrecognized option strings.»